src/Qowaiv/DecimalRound.cs

```diff
@@ -0,0 +1,206 @@
+﻿using System;
+
+namespace Qowaiv
+{
+    /// <summary>Extensions on <see cref="decimal"/> rounding.</summary>
+    public static class DecimalRound
+    {
+        private const int ScaleMask = 0x00FF0000;
+        private const int SignMask = unchecked((int)0x80000000);
+
+        public static decimal Round(this decimal value) => value.Round(0);
+        public static decimal Round(this decimal value, int decimals) => value.Round(decimals, DecimalRounding.AwayFromZero);
+        public static decimal Round(this decimal value, int decimals, int multiplyOf) => value.Round(decimals, DecimalRounding.AwayFromZero, multiplyOf);
+        public static decimal Round(this decimal value, int decimals, DecimalRounding rounding) => value.Round(decimals, rounding, 1);
+        public static decimal Round(this decimal value, int decimals, DecimalRounding rounding, int multiplyOf)
+        {
+            Guard.DefinedEnum(rounding, nameof(rounding));
+            Guard.Positive(multiplyOf, nameof(multiplyOf));
+
+            if ((decimals < -28) || (decimals > 28))
+            {
+                throw new ArgumentOutOfRangeException(nameof(decimals), QowaivMessages.ArgumentOutOfRange_DecimalRound);
+            }
+
+            var withMultiply = multiplyOf != 1;
+
+            var bits = decimal.GetBits(withMultiply ? value / multiplyOf : value);
+
+            int scale = (bits[3] & ScaleMask) >> 16;
+            var scaleDifference = scale - decimals;
+
+            if (scaleDifference <= 0)
+            {
+                return value;
```

Comment:
What if you have multiple of?

Reply:
It returns the value, and if `scaleDifference <= 0` than after the division, is still has no decimal digits that should not be there. However, I slightly changed this mechanism anyway.